### PR TITLE
improvement: schema validation error quality, phase 3

### PR DESCRIPTION
**Pull request by shockey**
### Description

Phase 3 of my schema validation improvements (phase 2: #1985).

This PR almost exclusively consists of moves from `oneOf`/`anyOf` to `switch` statements in the validation schema documents.

### How Has This Been Tested?

New unit tests were added to ensure that error quality won't erode over time.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/plugins/json-schema-validator/test-documents/response-schema-restrictions.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+cases:
```
Comment (shockey) [inline test/unit/plugins/json-schema-validator/test-documents/response-schema-restrictions.yaml]: needs a `meta` block
